### PR TITLE
[wip]Adding logs for requests

### PR DESCRIPTION
**Pull request by akhilkr128**
For the [feature](https://github.com/ManageIQ/manageiq/issues/21188)
Work in progress

### CODE REVIEW

File: lib/miq_automation_engine/engine/miq_ae_engine.rb [hunk 1]
```diff
@@ -67,45 +68,67 @@ def self.deliver_queue(args, options = {})
...
     options     = options_from_args(args)
     user_obj    = ae_user_object(options)
     state       = options[:state]
     vmdb_object = nil
     ae_result   = 'error'
     miq_task    = MiqTask.find(options[:open_url_task_id]) if options[:open_url_task_id]
-
+    $request_id = get_request_id(options)
+    $object_id = options[:object_id]
+    $object_type = options[:object_type]
```
Comment (Fryguy) [inline lib/miq_automation_engine/engine/miq_ae_engine.rb:94]: We shouldn't be using global vars here...perhaps they can be stored in the workspace?

File: lib/miq_automation_engine/engine/miq_ae_engine.rb [hunk 2]
```diff
@@ -114,17 +137,19 @@ def self.deliver(*args)
         else
           if ae_result.casecmp('error').zero?
             miq_task&.update_message(MiqTask::MESSAGE_TASK_COMPLETED_UNSUCCESSFULLY)
-            message = "Error delivering #{ManageIQ::Password.sanitize_string(options[:attrs].inspect)} for object [#{object_name}] with state [#{state}] to Automate: #{ws.root['ae_message']}"
+            message = "Error delivering #{attr_string} for object [#{object_name}] with state [#{state}] to Automate: #{ws.root['ae_message']}"
             _log.error(message)
+            RequestLogs.create(:log_message => message, :miq_requests_id => $request_id, :object_id => $object_id, :object_type => $object_type)
```
Comment (Fryguy) [inline lib/miq_automation_engine/engine/miq_ae_engine.rb:142]: I think we discussed this elsewhere, and I'm fine with this for now, but we might want to create a specialized RequestLogger that writes to the table as well as the log.

File: lib/miq_automation_engine/engine/miq_ae_engine.rb (hunk 1 above)
Comment (akhilkr128) [reply lib/miq_automation_engine/engine/miq_ae_engine.rb:94]: ok

File: lib/miq_automation_engine/engine/miq_ae_engine.rb (hunk 2 above)
Comment (akhilkr128) [reply lib/miq_automation_engine/engine/miq_ae_engine.rb:142]: Working on the logger.

**Review by Fryguy — CHANGES_REQUESTED** [review]
https://github.com/ManageIQ/manageiq-automation_engine/pull/503/files#r934935652 needs to be resolved before this can be merged.  A global variable there is a no-go because then we can't have multiple resolutions simultaneously.  I'm not sure what the exact right approach should be...perhaps it can be stored in the workspace or the miq_task?

File: lib/miq_automation_engine/engine/miq_ae_engine.rb [hunk 3]
```diff
@@ -67,45 +68,67 @@ def self.deliver_queue(args, options = {})
...
+  def self.get_request_id(options)
+    # binding.pry
+    if options[:object_type].ends_with?("Request")
+      return options[:object_id]
+    end
+    if options[:object_type].ends_with?("Task")
+      return MiqRequestTask.find(options[:object_id]).miq_request_id
+    end
+    if options[:object_type].starts_with?("Service")
+      return Service.find(264).miq_request.id
```
Comment (Fryguy) [inline lib/miq_automation_engine/engine/miq_ae_engine.rb:80]: This is hardcoding a specific service, so will need to be fixed before completion.

**Comment by Fryguy** [comment]
Closing in favor of https://github.com/ManageIQ/manageiq-automation_engine/pull/516

**Pull request closed without merging**